streams/src/test/java/org/apache/kafka/streams/KafkaStreamsTest.java

```diff
@@ -491,175 +585,115 @@ public void shouldThrowExceptionSettingUncaughtExceptionHandlerNotInCreateState(
 
     @Test
     public void shouldThrowExceptionSettingStateListenerNotInCreateState() {
-        globalStreams.start();
+        final KafkaStreams streams = new KafkaStreams(new StreamsBuilder().build(), props, supplier, time);
+        streams.start();
         try {
-            globalStreams.setStateListener(null);
+            streams.setStateListener(null);
             fail("Should throw IllegalStateException");
         } catch (final IllegalStateException e) {
             // expected
         }
     }
 
     @Test
-    public void testIllegalMetricsConfig() {
-        props.setProperty(StreamsConfig.METRICS_RECORDING_LEVEL_CONFIG, "illegalConfig");
-
+    public void shouldAllowCleanupBeforeStartAndAfterClose() {
+        final KafkaStreams streams = new KafkaStreams(new StreamsBuilder().build(), props, supplier, time);
         try {
-            new KafkaStreams(builder.build(), props);
-            fail("Should have throw ConfigException");
-        } catch (final ConfigException expected) { /* expected */ }
+            streams.cleanUp();
+            streams.start();
+        } finally {
+            streams.close();
+        }
+        streams.cleanUp();
```

Comment:
What's up with this last one? Should it be in the finally block?

Reply:
Yes!